dev-client/src/App.tsx

```diff
@@ -78,18 +79,22 @@ function App(): React.JSX.Element {
   return (
     <GestureHandlerRootView style={style}>
       <Provider store={store}>
-        <NativeBaseProvider theme={theme}>
-          <Portal.Host>
-            <NavigationContainer>
-              <GeospatialProvider>
-                <Toasts />
-                <HomeScreenContextProvider>
-                  <RootNavigator />
-                </HomeScreenContextProvider>
-              </GeospatialProvider>
-            </NavigationContainer>
-          </Portal.Host>
-        </NativeBaseProvider>
+        <BottomSheetModalProvider>
```

Comment:
We need this provider to exist above `NativeBaseProvider` because we have embedded our own selection components inside NativeBase modals, and those components use the Gorhom `BottomSheetModal`. Without a provider above the NativeBase code, it cannot display selection elements. This was the cause of the crash.